services/QuillLMS/app/workers/update_milestones_worker.rb

```diff
@@ -8,11 +8,11 @@ def perform(activity_session_uid)
     return unless activity_session
     # more milestones can be added here as relevant, for now this just checks to see if a Completed Diagnostic milestone needs to be created
 
-    if activity_session.state == 'finished' && activity_session.classroom_unit_id && activity_session.activity.activity_classification_id === 4
-      teacher_milestones = activity_session&.classroom_unit&.unit&.user&.milestones
-      if teacher_milestones && !teacher_milestones.find_by(name: 'Complete Diagnostic')
-        teacher_milestones.push(Milestone.find_by(name: 'Complete Diagnostic'))
-      end
-    end
+    return unless activity_session.state == 'finished' && activity_session.classroom_unit_id && activity_session.activity.activity_classification_id === 4
```

Comment:
Maybe break this up into multiple (and get rid of the js threequal)
```suggestion
    return unless activity_session.state == 'finished' 
    return unless activity_session.classroom_unit_id
    return unless activity_session.activity.activity_classification_id == 4
```